tests/test_utils.py

```diff
@@ -31,7 +33,29 @@ def test_get_subclasses():
     assert all(
         c in get_subclasses(MCKernel).values() for c in [UniformlyRandom, Metropolis]
     )
-    assert all(c in get_subclasses(ThermalKernel).values() for c in [Metropolis])
+    assert Metropolis in get_subclasses(ThermalKernel).values()
```

Comment:
Why not have a check for all base classes that uses derived_class_factory, for example, basis, bias and usher, etc?